File: 큐/BOJ_18258.cpp
#include<iostream>
#include<queue>
using namespace std;

queue<int> Q;
int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
     int N;
    cin >> N;

    while (N--)
    {
        string command;
        cin >> command;
        if (command.compare("push")==0)
        {
            int num;
            cin >> num;
            Q.push(num);
        }
        else if (command.compare("pop")==0)
        {
            if (Q.size() == 0)
                cout << -1 << "\n";
            else
            {
                cout << Q.front() << "\n";
                Q.pop();
            }
        }
        else if (command.compare("size")==0)
            cout << Q.size() << '\n';
        else if (command.compare("empty")==0)
        {
            if (Q.size() == 0)
                cout << 1 << '\n';
            else
                cout << 0 << '\n';
        }else if(command.compare("front")==0){
            if (Q.size() == 0)
                cout << -1 << '\n';
            else
                cout << Q.front() << '\n';
        }else{
             if (Q.size() == 0)
                cout << -1 << '\n';
            else
                cout << Q.back() << '\n';
        }
    }
}

File: 큐/BOJ_18258.java
package 큐;
import java.io.*;
import java.util.*;
public class BOJ_18258 {
    static int n;
    static String order;
    static int num;
    static int arr[];
    static int pointer=0;
    static int backPointer=0;
    public static void main(String[] args) throws Exception{
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb=new StringBuilder();
        n=Integer.parseInt(br.readLine());
        arr=new int[2000001];

        for(int i=0;i<n;i++){
            String s=br.readLine();
            st=new StringTokenizer(s);
            order=st.nextToken();
            if(order.equals("push")) {
                arr[pointer]=Integer.parseInt(st.nextToken());
                pointer++;
            }
            switch(order){
                case "pop":
                if(pointer-backPointer==0) sb.append(-1).append("\n");
                else {
                    sb.append(arr[backPointer]).append("\n");
                    backPointer++;
                }
                break;
                
                case "front":
                if(pointer-backPointer==0) sb.append(-1).append("\n");
                else {
                    sb.append(arr[backPointer]).append("\n");
                }
                break;

                case "back":
                if(pointer-backPointer==0) sb.append(-1).append("\n");
                else {
                    sb.append(arr[pointer-1]).append("\n");
                }
                break;

                case "size":
                sb.append(pointer-backPointer).append("\n");
                break;

                case "empty":
                if(pointer-backPointer==0) sb.append(1).append("\n");
                else sb.append(0).append("\n");
                break;


            }
        }

        System.out.print(sb);
        br.close();
    }
}
